### PR TITLE
Add changelog 4.6.0

### PR DESCRIPTION
**Pull request by willingc**
@minrk I took a pass at creating the changelog.

### CODE REVIEW

**Comment by takluyver** [comment]
I think it would be good to highlight #209, which I hope will reduce the number of cases where the kernel won't start because there's a `random.py` (or similar) module in the working directory.

**Comment by takluyver** [comment]
(I'm biased, though, since I did that)

**Comment by willingc** [comment]
No problem @takluyver. I only scanned PRs down to where 4.5.2 was cut so didn't see this PR until you mentioned it. Adding :smile:

File: docs/changelog.rst [hunk 1]
```diff
@@ -1,6 +1,37 @@
 Changes in IPython kernel
 =========================
 
+4.6
+---
+
+4.6.0
+*****
+
+`4.6.0 on GitHub <https://github.com/ipython/ipykernel/milestones/4.6.0>`__
```
Comment (minrk) [inline docs/changelog.rst]: milestone is 4.6, not 4.6.0

File: docs/changelog.rst [hunk 2]
```diff
@@ -1,6 +1,37 @@
...
+    - `transient: dict`
+- Add  a new dict, `transient`, to message spec for `publish`. For a display
+  data message, `transient` contains data that shouldn't be persisted to files
+  or documents. Add a `display_id` to this `transient` dict by
+  `display(obj, display_id=...)`
+- Add `ipykernel_launcher` module which removes the current working directory
+  from `sys.path` before launching the kernel. This helps to reduce the cases
+  where the kernel won't start because there's a `random.py` (or similar)
+  module in the current working directory.
+- Add busy/idle messages on IOPub during processing of every kernel request
```
Comment (minrk) [inline docs/changelog.rst]: s/every kernel request/aborted requests/

File: docs/changelog.rst [hunk 3]
```diff
@@ -1,6 +1,37 @@
...
+- Add active event loop setting to GUI, which enables the correct response
+  to IPython's `is_event_loop_running_xxx`
+- Add Python 3.6 support
```
Comment (minrk) [inline docs/changelog.rst]: Python 3.6 support isn't new in 4.6. This is just changes in the Travis setup, and probably doesn't need to be in the changelog

File: docs/changelog.rst [hunk 4]
```diff
@@ -1,6 +1,37 @@
...
+- Modify `OutStream` to inherit from `TextIOBase` instead of object to improve
+  API support and error reporting
+- Fix IPython kernel death messages at start, such as "Kernel Restarting..."
+  and "Kernel appears to have died", when parent-poller handles PID 1
+- Various bugfixes
```
Comment (minrk) [inline docs/changelog.rst]: Add:

> include IPython kernelspec in wheels to reduce reliance on "native kernel spec" in jupyter_client

File: docs/changelog.rst [hunk 5]
```diff
@@ -1,6 +1,37 @@
...
+4.6.0
+*****
+
+`4.6.0 on GitHub <https://github.com/ipython/ipykernel/milestones/4.6.0>`__
+
+- Add to API `DisplayPublisher.publish` two new fully backward-compatible
+  keyword-args:
+    - `update: bool`
+    - `transient: dict`
+- Add  a new dict, `transient`, to message spec for `publish`. For a display
```
Comment (minrk) [inline docs/changelog.rst]: I'd change this to

> support new `transient` key in display_data messages...

ipykernel doesn't define these things, the msg_spec in `jupyter_client` does.

File: docs/changelog.rst (hunk 1 above)
Comment (willingc) [inline docs/changelog.rst]: Good catch.

File: docs/changelog.rst (hunk 4 above)
Comment (willingc) [inline docs/changelog.rst]: Added.

**Comment by willingc** [comment]
This should have all the requested edits. Let me know if you would like any other changes. I added the kernelspec with wheels on line 27.

**Comment by minrk** [comment]
Thanks, @willingc!

**Pull request merged**